### PR TITLE
[REF] Fix issue with using array access tools on NULL values

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
This fixes some issues found by unit tests where by we are trying to use array access on NULL values

Before
----------------------------------------
Tests fail on PHP7.4

After
----------------------------------------
Tests pass on PHP7.4

tests affected

https://test.civicrm.org/job/CiviCRM-Core-Edge/513/CIVIVER=master,label=test-3/testReport/(root)/CRM_Contact_Page_AjaxTest/testGetDedupes/ https://test.civicrm.org/job/CiviCRM-Core-Edge/513/CIVIVER=master,label=test-3/testReport/api.v4.Entity/ParticipantTest/testGet/ https://test.civicrm.org/job/CiviCRM-Core-Edge/513/CIVIVER=master,label=test-3/testReport/(root)/CRM_Report_Form_TestCaseTest/testBadReportOutput_with_data_set__1/ https://test.civicrm.org/job/CiviCRM-Core-Edge/513/CIVIVER=master,label=test-3/testReport/(root)/CRM_Mailing_TokensTest/testTokensWithoutMailingJob_with_data_set__1/

### CODE REVIEW

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -3288,7 +3288,7 @@ public function tag(&$values) {
 
     $etTable = "`civicrm_entity_tag-" . uniqid() . "`";
 
-    if ($useAllTagTypes[2]) {
+    if (!empty($useAllTagTypes[2])) {
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php:3291]: ok - seems odd we never see enotices on this already though?

File: CRM/Contact/Page/AJAX.php [hunk 2]
```diff
@@ -770,7 +770,7 @@ public static function getDedupes() {
   public static function getSearchOptionsFromRequest() {
     $searchParams = [];
     $searchData = $_REQUEST['search'] ?? NULL;
-    $searchData['value'] = CRM_Utils_Type::escape($searchData['value'], 'String');
+    $searchData['value'] = CRM_Utils_Type::escape($searchData['value'] ?? NULL, 'String');
```
Comment (eileenmcnaughton) [inline CRM/Contact/Page/AJAX.php:773]: ok

File: CRM/Contribute/Form/Contribution/Confirm.php [hunk 3]
```diff
@@ -1886,7 +1886,7 @@ public function setFormAmountFields($priceSetID) {
       // contribution amount to be null, so that it will not show
       // contribution amount same as membership amount.
       //@todo - merge with section above
-      if ($this->_membershipBlock['is_separate_payment']
+      if (!empty($this->_membershipBlock['is_separate_payment'])
```
Comment (eileenmcnaughton) [inline CRM/Contribute/Form/Contribution/Confirm.php:1889]: Same question - why no enotices already?

File: CRM/Contact/BAO/Query.php (hunk 1 above)
Comment (seamuslee001) [reply CRM/Contact/BAO/Query.php:3291]: Yeh this one is the odd one i dunno why

File: CRM/Contribute/Form/Contribution/Confirm.php (hunk 3 above)
Comment (seamuslee001) [reply CRM/Contribute/Form/Contribution/Confirm.php:1889]: Seems like it was only added with php7.4 https://www.php.net/manual/en/migration74.incompatible.php 

"Array-style access of non-arrays ¶
Trying to use values of type null, bool, int, float or resource as an array (such as $null["key"]) will now generate a notice."

**Comment by seamuslee001** [comment]
@eileenmcnaughton I should add that all of these changes are backed up by unit tests so that gives us some confidence at least as they were all found by unit tests

File: CRM/Profile/Page/MultipleRecordFieldsListing.php [hunk 4]
```diff
@@ -53,7 +53,7 @@ public function getBAOName() {
    *   (reference) of action links
    */
   public function &links() {
-    if (!(self::$_links[$this->_pageViewType])) {
+    if (!(isset(self::$_links[$this->_pageViewType]))) {
```
Comment (eileenmcnaughton) [inline CRM/Profile/Page/MultipleRecordFieldsListing.php:56]: OK - I guess this is a bit different - !empty is closer to the original meaning

**Comment by eileenmcnaughton** [comment]
@seamuslee001 OK - I think that one I commented should be !empty not !isset but otherwise OK

**Comment by seamuslee001** [comment]
Updated now @eileenmcnaughton

**Pull request merged**